guildwars2/account.py

```diff
@@ -347,9 +347,20 @@ def refined_armor_scan(a, b=ids_refined_envoy_armor):
             storage_spaces = ("bank", "shared", "material storage")
             storage_spaces = OrderedDict(list(zip(storage_spaces, results)))
             characters = results[3]
+
         except APIError as e:
             return await self.error_handler(ctx, e)
 
+        try:
+            doc = await self.fetch_key(user, ["tradingpost"])
```

Comment:
We're already getting `doc` in the last try/except block, use that, check if it has tradingpost scope
```py
if "tradingpost" in doc["permissions"]:
```
and only then try to call the API